test/user_defined_output/fake_plugin/fake_user_defined_output.h

```diff
@@ -49,8 +50,10 @@ class FakeUserDefinedOutputPlugin : public UserDefinedOutputPlugin {
   absl::StatusOr<Envoy::ProtobufWkt::Any> getPerWorkerOutput() const override;
 
 private:
-  int data_called_ = 0;
-  int headers_called_ = 0;
+  mutable Envoy::Thread::MutexBasicLockable data_lock_;
```

Comment:
Using multiple locks tends to be error prone (makes the code harder to understand). Assuming we don't have a good reason to do this, can we do with a single mutex instead?